core/src/Kokkos_Macros.hpp

```diff
@@ -61,6 +61,14 @@
  *  KOKKOS_ENABLE_CUDA_UVM            Use CUDA UVM for Cuda memory space.
  */
 
+#define KOKKOS_VERSION_COMPARE(COMPARE, MAJOR, MINOR, PATCH) \
+  (KOKKOS_VERSION COMPARE((MAJOR)*10000 + (MINOR)*100 + (PATCH)))
```

Comment:
Somehow, I feel uncomfortable providing the comparison operator as an argument and would instead introduce `KOKKOS_VERSION_LTE` and `KOKKOS_VERSION_GTE` (or `KOKKOS_VERSION_LT` and `KOKKOS_VERSION_GT` but I think the equal part is used more).

Reply:
I think I agree with Daniel on this one.

Reply:
Updated